helpers.js

```diff
@@ -169,33 +174,24 @@ async function analyze(config) {
   }
 
   const client = new armlet.Client(armletOptions);
-  const analyzeOpts = {
-    _: config._,
-    debug: config.debug,
-    data: mythril.truffle2MythrilJSON(buildObj),
-    logger: config.logger,
-    style: config.style,
-    timeout: (config.timeout || 120) * 1000,
 
-    // FIXME: The below "partners" will change when
-    // https://github.com/ConsenSys/mythril-api/issues/59
-    // is resolved.
-    partners: ['truffle'],
-  };
+  // Get list of smart contract build json files from truffle build folder
+  const jsonFiles = await trufstuf.getTruffleBuildJsonFiles(config.contracts_build_directory);
 
-  analyzeOpts.data.analysisMode = analyzeOpts.mode || 'full';
+  const analysisResults = await doAnalysis(client, config, jsonFiles);
 
-  const issues = await client.analyze(analyzeOpts);
-  const formatter = getFormatter(analyzeOpts.style);
-  const esIssues = mythril.issues2Eslint(issues, buildObj, analyzeOpts);
-  esReporter.printReport(esIssues, solidityFile, formatter, analyzeOpts.logger.log);
+  const formatter = getFormatter(config.style);
+  
+  analysisResults.forEach(({issues, solidityFile, buildObj }) => {
+    const esIssues = mythril.issues2Eslint(issues, buildObj, config);
```

Comment:
`Usage: truffle run analyze [options] [*contract-name*]` - I've tested analyze with few commands and found  one problem with this syntax:
When I run `truffle run analyze --debug contractName` truffle config treats  `contractName` - as it is part of `debug` flag, not a parameter. So to make it work we would need to place options at the end of command like this: `truffle run analyze contractName --debug` then it `debug` flag is set to `true` not `contractName` value.

Reply:
Ok. `--debug` is not likely to be used so we can assume it will be added a the end, and perhaps should note that in the help. 

One other thing to keep in mind is that we have an issue open for the truffle people to integrate options into whatever "unified" scheme they want to come up with. So we may have to bring this up there if it isn't automatically addressed. See https://github.com/trufflesuite/truffle/issues/1474

Reply:
I left detailed comment in  trufflesuite/truffle#1474 and asked for help in gitter chat too. Hopefully they will respond soon

Reply:
> If one of the analyses fails, what happens here?

Pushed changes and test that handles this case.

> Ok. `--debug` is not likely to be used so we can assume it will be added a the end,

Pushed changes for this too.